src/account.ts

```diff
@@ -0,0 +1,843 @@
+
+      // Greatest claim value/total amount we've sent the receiver
+      const spent = new BigNumber(this.account.bestOutgoingClaim ? this.account.bestOutgoingClaim.value : 0)
+      const remainingInChannel = channel.value.minus(spent)
+
+      if (remainingInChannel.lte(0)) {
+        this.master._log.trace(`Cannot send claim to ${this.account.accountName}: no remaining funds in outgoing channel`)
+        return settlementBudget
+      }
+
+      if (settlementBudget.lte(0)) {
+        this.master._log.trace(`Cannot send claim to ${this.account.accountName}: no remaining settlement budget`)
+        return Promise.reject(settlementBudget)
+      }
+
+      // Ensures that the increment is greater than the previous claim
+      // Since budget and remaining in channel must be positive, claim increment should always be positive
+      const claimIncrement = BigNumber.min(remainingInChannel, settlementBudget)
+      // Total value of new claim: value of old best claim + increment of new claim
+      const value = spent.plus(claimIncrement)
+
+      const channelId = channel.channelId
+      const contract = await getContract(this.master._web3)
+      const digest = await contract.methods.paymentDigest(channelId, value.toString()).call()
+      const signature = await this.master._web3.eth.sign(digest, channel.sender)
+
+      const claim = {
+        channelId,
+        signature,
+        value: new BigNumber(value).toString()
+      }
+
+      this.master._log.trace(`Sending claim for total of ${format(value, Unit.Wei)}, ` +
+        `incremented by ${format(claimIncrement, Unit.Wei)}, to ${this.account.accountName}`)
+      this.account.bestOutgoingClaim = claim
+
+      // Since claimIncrement <= settlementBudget, new budget should always be >= 0
+      settlementBudget = settlementBudget.minus(claimIncrement)
+
+      // Send paychan claim to client, don't await a response
+      this.sendMessage({
+        type: BtpPacket.TYPE_TRANSFER,
+        requestId: await requestId(),
+        data: {
+          amount: convert(value, Unit.Wei, Unit.Gwei).toFixed(0, BigNumber.ROUND_CEIL),
+          protocolData: [{
+            protocolName: 'machinomy',
+            contentType: BtpPacket.MIME_APPLICATION_JSON,
+            data: Buffer.from(JSON.stringify(claim))
+          }]
+        }
+      }).catch(err => {
+        // In any case, this isn't particularly important/actionable
+        this.master._log.trace(`Error while sending claim to peer: ${err.message}`)
+      })
+
+      return settlementBudget
+    } catch (err) {
+      this.master._log.error(`Failed to send claim to ${this.account.accountName}: ${err.message}`)
+      throw settlementBudget
+    }
+  }
+
+  async handleData (message: BtpPacket, dataHandler?: DataHandler): Promise<BtpSubProtocol[]> {
+    const info = getSubprotocol(message, 'info')
+    const ilp = getSubprotocol(message, 'ilp')
```

Comment:
nit: don't need to check for both if it's only valid to have included one of them (and one will be ignored if you set both anyway)

Reply:
Yeah, it'd be weird for the packet to include both. I think having both checks is useful though in case they sent some random subprotocol name that we don't support.